awx/main/models/credential/__init__.py

```diff
@@ -504,6 +509,15 @@ class Meta:
                     'Ansible Tower documentation for example syntax.')
     )
 
+    @classmethod
+    def from_db(cls, db, field_names, values):
```

Comment:
I had several times looked into `from_db` and shied away from it. The best summary of the situation comes from a StackOverflow comment:

https://stackoverflow.com/questions/1355150/django-when-saving-how-can-you-check-if-a-field-has-changed#comment82360065_31202901

> you still need to handle 3 cases: __init__/from_db, refresh_from_db, and save

I get that you're probably safe from these concerns in this particular implementation, because inputs and injectors for the managed types are fundamentally static. There's no foreseeable modification of these fields... as long as `refresh_from_db` doesn't reset it back to `{}` or something like that.

I vaguely remember some other concerns, that some managers might not abide by the from_db method, but I don't remember the specifics.

Reply:
From my testing, it _does_ seem to work reliably well for our needs (listing/filtering the API and having the serializer show the correct data).